go/service/gregor.go

```diff
@@ -96,6 +146,10 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
 
+	// Send message to local state machine
+	g.gregorCli.ConsumeMessage(m)
```

Comment:
i wonder if you can dangle a boolean off of `context.Context` to say not to bother shipping back to the state machine if this comes from `sync`?